File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       
        
        int total = nums[0];
        int max = total;
        for(int i = 1; i < nums.length; i++){
            if(total < 0 && nums[i] > total){
                total = nums[i];
            }else{
                total += nums[i];
            }
            max = Math.max(total,max);
        }
        
        return max;
        
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        dp = []
        

        dp.append(nums[0])
        curr = nums[0]
        
        for i in range(1, len(nums)):
            
            dp.append(max(nums[i], dp[i-1] + nums[i]))
            curr = max(curr, dp[i])
        
        return curr
           
            
            
                
                
            